hack/metadata-lint.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -x
+
+# Ensure the enhancement metadata includes the required information
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# We only look at the files that have changed in the current PR, to
+# avoid problems when the template is changed in a way that is
+# incompatible with existing documents.
+CHANGED_FILES=$(git log --name-only --pretty= "${PULL_BASE_SHA:-origin/master}.." -- \
+                    | (grep '^enhancements.*\.md$' || true) \
+                    | sort -u)
+
+(cd tools && go build -o ../enhancement-tool -mod=mod ./main.go)
```

Comment:
We should ideally build the binary in the Dockerfile and just run it here.

Reply:
Ideally, yeah. Right now that Dockerfile is used for several things, each of which is installed separately. So, I was following the existing pattern. Maybe we can review that separately.

Reply:
That works for me.